tower/src/limit/concurrency/layer.rs

```diff
@@ -22,3 +25,27 @@ impl<S> Layer<S> for ConcurrencyLimitLayer {
         ConcurrencyLimit::new(service, self.max)
     }
 }
+
+/// Enforces a limit on the concurrent number of requests the underlying
+/// service can handle.
+/// This variant accepts a owned semaphore (`Arc<Semaphore>`) which can
+/// be reused across services.
```

Comment:
The word "variant" makes a think of an enum but I don't have any better suggestions 🤷

Reply:
how about just
```suggestion
/// Unlike [`ConcurrencyLimitLayer`], which enforces a per-service concurrency
/// limit, this layer accepts a owned semaphore (`Arc<Semaphore>`) which can be
/// shared across multiple services.
```
or something?